duck/utils/redefine_HB_pair.py

```diff
@@ -0,0 +1,98 @@
+
+def find_ligand(system, ligand_name = 'UNL'):
+    for r in system.residues:
+        if r.name == ligand_name:
+            return r
+    raise ValueError('Could not find the ligand in the system with the given name.')
+
+def find_interacting_atoms(ref, ligand, allowed_elements = [7,8], distance_threshold = 3.5, pocket_points=None, pocket_radius=1):
+    r = []
+    ref_coords = (ref.xx, ref.xy, ref.xz)
+    for atom in ligand.atoms:
+        lig_atom_coords = (atom.xx, atom.xy, atom.xz)
+        #distance to reference
+        dist = math.dist(ref_coords, lig_atom_coords)
+        if atom.element in allowed_elements and dist <= distance_threshold:
+            # in pocket
+            if pocket_points:
+                for pocket_atom in pocket_points:
+                    pocket_atom_coords = (pocket_atom.xx, pocket_atom.xy,pocket_atom.xz)
+                    if math.dist(pocket_atom_coords, lig_atom_coords) <= pocket_radius:
+                        r.append((atom, dist))
+                        break
+            else:
+                r.append((atom, dist))
+    if len(r) == 0: raise ValueError('No atom fits the interacting criteria.')
+    return r
+def modify_DUck_HB_atom_ids(prot_atom, lig_atom, files_to_mod = ['dist_duck.rst', 'dist_md.rst'], backup=True):
+    new_prot_idx = str(int(prot_atom.idx)+1)
+    new_lig_idx = str(int(lig_atom.idx)+1)
+    for file in files_to_mod:
+        if backup and not os.path.exists(f'{file}.BAK'):
+            shutil.copyfile(file, f"{file}.BAK")
+        with FileInput(files=file, inplace=True) as f:
+            for line in f:
+                line = line.rstrip()
+                if 'iat' in line:
+                    line_parts = line.split()
+                    mod_parts = []
+                    for part in line_parts:
+                        if 'iat' in part:
+                            part = f'iat={new_prot_idx},{new_lig_idx},'
+                        mod_parts.append(part)
+                    print(' '.join(mod_parts))
+                else: print(line)
+                    
+def find_and_modify_HB_interaction(system_pmd, interaction, ligand_name, distance_threshold=3.5, pocket_points =None, pocket_points_radius=0, modify=False, files_to_modify = ['dist_duck.rst', 'dist_md.rst']):
+    prot_atom = find_protein_atom(system_pmd, interaction)
+    ligand = find_ligand(sys_pmd, ligand_name)
+    int_atoms = sorted(find_interacting_atoms(prot_atom, ligand, distance_threshold=distance_threshold, pocket_points=pocket_points, pocket_radius=pocket_points_radius),key=lambda x:x[1])
+    good_lig_atom = int_atoms[0]
+    if args.modify: modify_DUck_HB_atom_ids(prot_atom, good_lig_atom[0], files_to_mod = files_to_modify, backup=True)
+    return good_lig_atom
+
+if __name__=='__main__':
+    
+    parser = argparse.ArgumentParser(description='Helper tool to redefine the hydrogen bond ligand atom and modify the amber inputs for DUck.')
```

Comment:
could you maybe extend this with 1-2 more sentences? It's not immediately clear what the purpose of the script is and where it should be applied in the pipeline